### PR TITLE
Adjust to more user favorable column widths

### PR DESCRIPTION
**Pull request by kingthorin**
Modify `ParamsPanel.setParamsTableColumnSizes()` loop through all headings and set the preferred width based on the header content. After the loop set the "value" column to be preferred much larger than it's header.

### CODE REVIEW

File: src/org/zaproxy/zap/extension/params/ParamsPanel.java [hunk 1]
```diff
@@ -204,23 +204,26 @@ private JScrollPane getJScrollPane() {
...
 		
-		paramsTable.getColumnModel().getColumn(2).setMinWidth(50);
-		paramsTable.getColumnModel().getColumn(2).setPreferredWidth(100);	// used
+		paramsTable.getColumnModel().getColumn(2).setMinWidth(40);
+		paramsTable.getColumnModel().getColumn(2).setPreferredWidth(40);	// used
 		
-		paramsTable.getColumnModel().getColumn(3).setMinWidth(50);
-		paramsTable.getColumnModel().getColumn(3).setPreferredWidth(100);	// numvals
+		paramsTable.getColumnModel().getColumn(3).setMinWidth(40);
+		paramsTable.getColumnModel().getColumn(3).setPreferredWidth(40);	// numvals
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/params/ParamsPanel.java]: This hides part of the column name even if there's space, was that on purpose? (First time users would have to resize the column to know what values the column has.)

File: src/org/zaproxy/zap/extension/params/ParamsPanel.java [hunk 2]
```diff
@@ -204,23 +204,26 @@ private JScrollPane getJScrollPane() {
...
 		
-		paramsTable.getColumnModel().getColumn(4).setMinWidth(50);
-		paramsTable.getColumnModel().getColumn(4).setPreferredWidth(100);	// % change
+		paramsTable.getColumnModel().getColumn(4).setMinWidth(40);
+		paramsTable.getColumnModel().getColumn(4).setPreferredWidth(40);	// % change
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/params/ParamsPanel.java]: Ditto.

File: src/org/zaproxy/zap/extension/params/ParamsPanel.java (hunk 1 above)
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/params/ParamsPanel.java]: Tweaked

File: src/org/zaproxy/zap/extension/params/ParamsPanel.java (hunk 2 above)
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/params/ParamsPanel.java]: Tweaked

File: src/org/zaproxy/zap/extension/params/ParamsPanel.java [hunk 3]
```diff
@@ -204,23 +204,26 @@ private JScrollPane getJScrollPane() {
...
+		paramsTable.getColumnModel().getColumn(0).setPreferredWidth(60);	// type
 		
 		paramsTable.getColumnModel().getColumn(1).setMinWidth(100);
-		paramsTable.getColumnModel().getColumn(1).setPreferredWidth(200);	// name
+		paramsTable.getColumnModel().getColumn(1).setPreferredWidth(100);	// name
 		
-		paramsTable.getColumnModel().getColumn(2).setMinWidth(50);
-		paramsTable.getColumnModel().getColumn(2).setPreferredWidth(100);	// used
+		paramsTable.getColumnModel().getColumn(2).setMinWidth(40);
+		paramsTable.getColumnModel().getColumn(2).setPreferredWidth(40);	// used
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/params/ParamsPanel.java]: This column is now trimmed. For example, in French this column and "% Change" are trimmed as well.

May I suggest removing the hardcoded sizes and pack the columns? Didn't try it but based on other tables it should look fine in more languages, although we would have to take care to not use the whole value for the last column, given how bigger the values are.

**Comment by kingthorin** [comment]
@thc202 yes the widths only really consider English, as with the originals. Can you point me to a packed example? I did try a few other variations that didn't depend on statically defining things but then the columns didn't seem to fill the width available, and it looked funny to have unused space.

**Comment by kingthorin** [comment]
Hopefully that's a more i18n friendly implementation 👍

**Comment by thc202** [comment]
Sorry for not answering in time, it could be used the methods of `JXTable` to pack the columns (which do a similar thing to what is being done here).

**Comment by thc202** [comment]
Thank you!

**Comment by kingthorin** [comment]
Should I look into changing it, or leave it?

**Comment by thc202** [comment]
Leave it :)

**Comment by kingthorin** [comment]
Ok @psiinon, look good to u?

**Comment by psiinon** [comment]
yep :)

**Comment by kingthorin** [comment]
Thanks

**Pull request merged**